### ISSUE TITLE
Error with ACCOUNT_EMAIL_VERIFICATION = 'mandatory' and ACCOUNT_CONFIRM_EMAIL_ON_GET = True

### ISSUE THREAD

**Issue by brunomichetti** [jazzband/dj-rest-auth#138]
Hello I'm using `dj-rest-auth` in my app with mandatory email account confirmation, and  account confirmation when click on the link received in the email.

Doing the GET to the generated link this error raises:

```
TemplateResponseMixin requires either a definition of 'template_name' or an implementation of 'get_template_names()'
```

Here I show you a screenshot:
![Screen Shot 2020-09-02 at 15 49 14](https://user-images.githubusercontent.com/20806762/92023963-e50a6480-ed33-11ea-89d3-502d0dd016ef.png)

I've found a solution, that is adding this in my urls:

```
path(
        'rest-auth/registration/account-confirm-email/<str:key>/',
        ConfirmEmailView.as_view(),
    )
```

It works if I add that path `BEFORE` the `path('rest-auth/registration/', include('dj_rest_auth.registration.urls')),`. Otherwise it raises the mentioned error.

Is there a better solution? Do you know how can be this issue solved?

**Comment by jerinpetergeorge** [comment]
Could you add the **`ConfirmEmailView`** class? It seems like the issue belongs to your view class itself. @brunomichetti

**Comment by brunomichetti** [comment]
> Could you add the **`ConfirmEmailView`** class? It seems like the issue belongs to your view class itself. @brunomichetti

Thanks @jerinpetergeorge for the response.
ConfirmEmailView is the one from dj_rest_auth. I import the view from `dj_rest_auth.registration.views`.
I think the problem is in that view class too

**Comment by jerinpetergeorge** [comment]
from the [codebase](https://github.com/jazzband/dj-rest-auth/blob/master/dj_rest_auth/registration/urls.py#L15)

> ```
>     # account_confirm_email - You should override this view to handle it in
>     # your API client somehow and then, send post to /verify-email/ endpoint
>     # with proper key.
>     # If you don't want to use API on that step, then just use ConfirmEmailView
>     # view from:
>     # django-allauth https://github.com/pennersr/django-allauth/blob/master/allauth/account/views.py
>     re_path(r'^account-confirm-email/(?P<key>[-:\w]+)/$', TemplateView.as_view(),
>         name='account_confirm_email'),
> ```

So, when you are using *`path('rest-auth/registration/', include('dj_rest_auth.registration.urls'))`* in your code, the `dj-rest-auth` will use the Django's [**`TemplateView`**](https://docs.djangoproject.com/en/dev/ref/class-based-views/base/#django.views.generic.base.TemplateView) class for the *account confirmation*.  But, in your case, you are using **`ConfirmEmailView`** which is *originally* defined in the [`all-auth` package](https://github.com/pennersr/django-allauth/blob/32b42dfdbfb184f40f548f3ffccc58239e024a02/allauth/account/views.py#L287). @brunomichetti

**Comment by brunomichetti** [comment]
> from the [codebase](https://github.com/jazzband/dj-rest-auth/blob/master/dj_rest_auth/registration/urls.py#L15)
> 
> > ```
> >     # account_confirm_email - You should override this view to handle it in
> >     # your API client somehow and then, send post to /verify-email/ endpoint
> >     # with proper key.
> >     # If you don't want to use API on that step, then just use ConfirmEmailView
> >     # view from:
> >     # django-allauth https://github.com/pennersr/django-allauth/blob/master/allauth/account/views.py
> >     re_path(r'^account-confirm-email/(?P<key>[-:\w]+)/$', TemplateView.as_view(),
> >         name='account_confirm_email'),
> > ```
> 
> So, when you are using _`path('rest-auth/registration/', include('dj_rest_auth.registration.urls'))`_ in your code, the `dj-rest-auth` will use the Django's [**`TemplateView`**](https://docs.djangoproject.com/en/dev/ref/class-based-views/base/#django.views.generic.base.TemplateView) class for the _account confirmation_. But, in your case, you are using **`ConfirmEmailView`** which is _originally_ defined in the [`all-auth` package](https://github.com/pennersr/django-allauth/blob/32b42dfdbfb184f40f548f3ffccc58239e024a02/allauth/account/views.py#L287). @brunomichetti

Yes, but the registration should work without the need of adding that path, shouldn't?

**Comment by jerinpetergeorge** [comment]
As the comment indicates, if you are using the `dj_rest_auth.registration.urls` for the *"account confirmation"*, the process will not get completed as the URL pointing to a *dummy view*, which does nothing.

**Comment by brunomichetti** [comment]
> As the comment indicates, if you are using the `dj_rest_auth.registration.urls` for the _"account confirmation"_, the process will not get completed as the URL pointing to a _dummy view_, which does nothing.
> 
> ### Update
> @brunomichetti I have managed to recreate the issue.
> 
> If you are getting an exception as
> 
> > TemplateResponseMixin requires either a definition of 'template_name' or an implementation of 'get_template_names()'
> 
> which means you are still requesting to the rest auth's account-confirm-email view, not to the **`ConfirmEmailView`**.

Thank you for the help.
Don't you think that if you set `ACCOUNT_EMAIL_VERIFICATION = 'mandatory'` and `ACCOUNT_CONFIRM_EMAIL_ON_GET = True` the get to the generated url should validate and return a valid answer without the need of adding that path?

**Comment by jerinpetergeorge** [comment]
Those settings are irrelevant here. 

You can try any arbitrary key (ex: `/rest-auth/registration/account-confirm-email/foobar/`) to access the URL and it will give you some other response as below,

![image](https://user-images.githubusercontent.com/21334929/100468420-68044680-30fa-11eb-9fbe-a2fbc13663c7.png)

**Comment by brunomichetti** [comment]
> Those settings are irrelevant here.
> 
> You can try any arbitrary key (ex: `/rest-auth/registration/account-confirm-email/foobar/`) to access the URL and it will give you some other response as below,
> 
> ![image](https://user-images.githubusercontent.com/21334929/100468420-68044680-30fa-11eb-9fbe-a2fbc13663c7.png)

Sorry, but I don't understand what you mean. I want to have:
1. set email confirmation as mandatory
2. if I do a GET to the url received on the email then receive a valid response, not the error.

Is this right now possible with dj-rest-auth without the need of adding the path?

**Comment by jerinpetergeorge** [comment]
No, that is not possible **_without adding the path_**

**Comment by brunomichetti** [comment]
> No, that is not possible **_without adding the path_**

Do you agree with me that a fix should be done to avoid the adding of the path?

**Comment by jerinpetergeorge** [comment]
That's a nice suggestion!!!

**Comment by jerinpetergeorge** [comment]
I have created a separate issue [here](https://github.com/jazzband/dj-rest-auth/issues/180). Let's hope we will be able to see it on the master branch soon.

**Comment by brunomichetti** [comment]
> I have created a separate issue [here](https://github.com/jazzband/dj-rest-auth/issues/180). Let's hope we will be able to see it on the master branch soon.

Thanks a lot!!!! 
Hope this can be addressed soon
